tinnsleep/utils.py

```diff
@@ -105,6 +106,7 @@ def merge_labels_list(list_valid_labels, n_epoch_final, merge_fun=np.all):
                                       resampling_factor == 1 for j in range(n_epoch_final)]
 
         else:
+            logger.warning("Interpolating non proportional list, expecting to have non-uniform shift across recording")
```

Comment:
Doesn't work!!
This "else" mention concerns all the cases but the one where we are moving from small epochs to larger epochs. In our case, we don't want to write this warning if we are moving from big epochs to smaller epochs with proportionality, which it does currently!
If you want this warning to work, you have to write:

`        else:
            if not float(n_epoch_final / n_epoch_before).is_integer():
                       logger.warning("Interpolating non proportional list, expecting to have non-uniform shift across recording")`

Reply:
the warning works on my side.
`merge_labels_list([[True, False, True]], 2)
WARNING:root:Interpolating non proportional list, expecting to have non-uniform shift across recording`

Reply:
Yes but if you try:
`merge_labels_list([[True, True]], 4)`

you shouldn't have a warning, right?
Try it and see if you don't get the warning ;)...

Reply:
@RobinGuillard it is exactly what I'm fixing now (resampling factor is inverted and upsampling is not managed)